more structured quick start (#542)
applegrew/django-select2: docs/get_started.rst

.. automodule:: django_select2
    :members:

Assumptions
-----------

> * You have a Django project, possibly with models linked by foreign key fields,

Reviewer: That is the main scenario, but not all of them. Maybe it makes sense to coin it, so it is clear that the library does support other scenarios.

Author: I"m not sure I understand your comment.  what about you edit the assumptions as to make them more correct, after you accept the PR?

> * You have a Django project, possibly with models linked by foreign key fields,

Reviewer: ```suggestion
* You have a running Django project, possibly with models linked by foreign key fields,
```

* are using Python3,
* set up a virtual environment with ``python3 -m venv <location>``,
* activated the virtual environment,

> * installed ``django`` using ``pip``,

Reviewer: Django is a dependency of this package. So you don't necessarily need to install it. However, you do need an app setup.
Also, you may use `pipenv`, `pip` is not hard requirement.

Author: I hadn't ever heard of `pipenv`, but I read »The official recommended way [to install pipenv] is to use pip«.  

assumption: »you have a Django app«?

Reviewer: `Pipenv` is a whole toolset. You can use it to describe the entire environment including checksums, to ensure your environments are the same across machines. You should try it out!

Besides that, maybe better: you have a **running** Django  app

0. Follow the installation instructions above.

1. Replace native Django forms widgets with one of the several ``django_select2.form`` widgets.

>    Start by importing them into your forms.py, right next to django own ones::

Reviewer: ```suggestion
   Start by importing them into your ``forms.py``, right next to Django own ones::
```

     from django import forms

>      from django_select2 import forms as s2forms

Reviewer: Good call to include the imports. I always forget 👍

>    Then let's assume you have a model with a Choice, a ForeignKey, and a

Reviewer: ```suggestion
   Then let's assume you have a model with a choice, a :class:`.ForeignKey`, and a
```

>    ManyToMany field, you would add this information to your Form Meta

Reviewer: ```suggestion
   :class:`.ManyToManyField`, you would add this information to your Form Meta
```

   class::

>         widgets = {

Reviewer: Please include the entire code example, otherwise people will be confused.

Author: I haven't written it yet.  I'll review the PR in its entirety, and come back to you soon (36h).

Author: 36h went by, and I'm still busy on the project I'm working at. (mfrasca/ghini)
sooner or later, from that project, I will again need to dive into djangoSelect2, and I will again read the docs and amend the docs where what I will be reading doesn't make sense to me.
in particular, I will need the Multiple select widget, which I've not yet used.

            'category': s2forms.Select2Widget,

>             'author': s2forms.ModelSelect2Widget(model=auth.get_user_model(),

Reviewer: You did not import `auth`, I would also suggest to import `get_user_model` directly instead, since you don't use anything else from the `auth` module.
